compiler/optimizer/OMRLocalCSE.cpp

```diff
@@ -777,8 +777,11 @@ void OMR::LocalCSE::doCommoningIfAvailable(TR::Node *node, TR::Node *parent, int
        shouldCommonNode(parent, node) &&
        performTransformation(comp(), "%s   Local Common Subexpression Elimination commoning node : %p by available node : %p\n", optDetailString(), node, availableExpression))
       {
+      // We can allow final non volatile fields to be commoned even during the volatile only phase
+      // as we do not expect those field to be changing. This enables up to common volatiles that are based on an 
+      // indirection chain of such final non volatiles (or autos or parms that are not global by definition)
       if (!node->getOpCode().hasSymbolReference() ||
-          (_volatileState == VOLATILE_ONLY && node->getSymbol()->isVolatile()) ||
+          (_volatileState == VOLATILE_ONLY && (node->getSymbol()->isVolatile() || node->getSymbol()->isAutoOrParm())) ||
```

Comment:
I think this question of what non-volatiles that can common should be pulled into an API so it can be replaced/modified by downstream projects.

Reply:
I have added the query to OMRLocalCSE opt in https://github.com/eclipse/omr/compare/10527798af1e29f001de3b99d9ef0e3d1240de19..4c9c4fca399f6bdef3baca5dd21477cf847d30bd which we can override in https://github.com/eclipse/openj9/blob/master/runtime/compiler/optimizer/J9LocalCSE.cpp